### DISCUSSION TITLE
App icons for smoother transition

### DISCUSSION THREAD

**Discussion by pointlessone** [zed-industries/zed#13220] [Show and tell]
I'm trying Zed as my main editor. I like it a lot but I've noticed an issue: I'm losing the app in the dock and app switcher all the time. Apparently, after years of MacVim I became conditioned to associate color (green) with my code editor. It didn't help that Zed's black icon doesn't particularly pop when macOS is in dark mode.

So I made myself an icon that would look more like what I'm used to. It worked great for me and I made a bunch more icons. Here they are. Hope they will help others switch smoother.

| | |
|-|-|
| Emacs | <img width="256" alt="Emacs Zed Icon" src="https://github.com/zed-industries/zed/assets/7405/74fd4ce4-9692-4373-9ebe-9ed418a696bb"> |
| MacVim | <img width="256" alt="MacVim Zed Icon" src="https://github.com/zed-industries/zed/assets/7405/e3fafde0-be5a-437a-bc91-675d805a1689"> |
| Sublime Text (variant 1) | <img width="256" alt="Sublime Text Zed Icon, variant 1" src="https://github.com/zed-industries/zed/assets/7405/76ca4f50-5cbe-40fb-b394-b95e101ac529"> |
| Sublime Text (variant 2) | <img width="256" alt="Sublime Text Zed Icon, variant 2" src="https://github.com/zed-industries/zed/assets/7405/7af558a0-03ad-44f4-b558-39647afe1578"> |
| vim | <img width="256" alt="vim Zed Icon" src="https://github.com/zed-industries/zed/assets/7405/19dfebf5-765d-4ac1-8f41-7b8b00534ea6"> |
| VS Code | <img width="256" alt="VS Code Zed Icon" src="https://github.com/zed-industries/zed/assets/7405/9f4c8aef-571a-4b09-987e-08cc6627f0e6"> |

And a few color variations.

| | |
|-|-|
| Deep Green | <img width="256" alt="Deep Green Zed Icon" src="https://github.com/zed-industries/zed/assets/7405/5553fbb1-05d3-4a35-94f8-f748da91dbf2"> |
| White | <img width="256" alt="White Zed Icon" src="https://github.com/zed-industries/zed/assets/7405/ae2e81a6-47e8-4aab-a2d2-eba6fe38c36c"> |
| Black | <img width="256" alt="Black Zed Icon" src="https://github.com/zed-industries/zed/assets/7405/0638c3f6-f93b-4140-b13a-b5b62af8edc7"> 

Black is basically a recreation of the original Zed icon. I didn't find the source file I could edit so I first recreated it in Sketch so that I could make the one I actually wanted and it looked similar to the original. I don't know why anyone would go to the trouble of changing original to this one but since I had it I threw it in, too.

All the icons: [Icons.zip](https://github.com/user-attachments/files/15889960/Icons.zip)

## HOWTO change Zed icon in macOS

1. Find Zed in your `/Applications` directory and open context menu. Either right-click it, of Control-click the app to open the menu. Pick Get Info.

   <img width="344" alt="image" src="https://github.com/zed-industries/zed/assets/7405/b999f3a2-f6af-44ce-9753-df1a51d7f1a6">

2. Pick an icon you like and drop it onto the app icon at the top of the Info window.

   <img width="315" alt="Screenshot 2024-06-18 at 20 07 06" src="https://github.com/zed-industries/zed/assets/7405/e906f116-531f-4716-a1f3-be0dbffc0ee9">

3. You're done. You might need to restart Zed to see the new icon in the Dock and App Switcher.

**Comment by nstlgy** [comment]
Theyre pretty cool, i myself like the Sublime Text (variant 2) alot! Thanks for sharing

**Comment by d1y** [comment]
Thank you for sharing

**Comment by jknlsn** [comment]
This is cool as! Any chance you could share the Sketch file, or export as SVG for editing?

I'd love some tweaks along the lines of these which match popular themes here https://github.com/dhanishgajjar/terminal-icons, even just for personal use I wouldn't mind creating myself!

**Reply by pointlessone** [reply]
Here you go!
[Icon.sketch.zip](https://github.com/user-attachments/files/16179302/Icon.sketch.zip)

**Reply by jknlsn** [reply]
Thank you!

**Reply by ma-zzini** [reply]
Hi can you share also the svg file ? Thanks a ton!

**Reply by pointlessone** [reply]
Here's an export but it doesn't look quite right. Specifically, it misses a mask on the logo. It's supposed to be a conical gradient which SVG doesn't support. Inkscape has some hints on how to mimic it: https://wiki.inkscape.org/wiki/index.php/Advanced_Gradients#Conical_gradient. The noise texture probably can be done better in SVG, too.

I'm afraid you'll have to fix it yourself.

![icon_512x512@2x@2x](https://github.com/user-attachments/assets/ae878241-7de9-4991-8eff-39059b51ff1d)

**Comment by MihaelBercic** [comment]
Thanks for sharing!

**Comment by saminiemi** [comment]
I would be nice to have these or similar icons to the core. While it is easy to change app icon as described above, you have to do that every time app updates, which it very often in case of Zed.

**Comment by PixelCookie-zyf** [comment]
It's a really great sharing. But there is still a problem that every time Zed is updated, it causes the Zed icon to be disordered, and then we need to operate it again. Is there a solution to this?

**Reply by pointlessone** [reply]
I use [fileicon](https://github.com/mklement0/fileicon) to set the icon after upgrade. I use it with an FS monitoring tool to run the command when `/Applications/Zed.app` changes (there are a plenty: `watchman` is popular, there's also [Hazel](https://www.noodlesoft.com/) for a nice GUI, and I'm sure many more).

**Reply by PixelCookie-zyf** [reply]
Thanks, I will give it a try.

**Comment by ladusee** [comment]
It would be nice to have both light (white) and dark icons, switching automatically to match the user’s system theme. Or manually. Light icon might be with the blue accent, like on the website.

**Comment by jethroo** [comment]
exactly what I was looking for, many thanks! ❤️
